JSTOR-Elasticsearch-with-embeddings.py

```diff
@@ -0,0 +1,540 @@
+import elasticsearch
+import elasticsearch.helpers
+import json
+from enum import Enum
+from scipy.spatial.distance import cosine
+
+import torch
+from transformers import BertTokenizer, BertModel
+
+# CONFIGS
+DATASET_DIR = "/scratch/avtale/jstor/letter_fgh" #root directory for files
+ES_INDEX_NAME = "jstor_articles_w_contextual_embeddings"
+ES_HOST = "agave1.agave.rc.asu.edu:9200"
+MAPPINGS_JSON = "data/mapping_beckett_jstor"
+LOG_PREFIX = "ace"
+
+ES_AUTH_USER = None
+ES_AUTH_PASSWORD = None
+
+ES_DOCUMENT_TYPE = 'article'
+ES_CREATE_INDEX = True
+ES_TIMEOUT = 60
+
+DEBUG = ''
+KEY_SPECIES="species"
+KEY_SUBSPECIES="subspecies"
+
+
+# LOGGING
+logging.getLogger().setLevel(logging.DEBUG)
+logger = logging.getLogger('jstor')
+DEBUG_LOG_FILENAME = '{}_debug.log'.format(LOG_PREFIX)
+WARN_LOG_FILENAME = '{}_warn.log'.format(LOG_PREFIX)
+formatter = logging.Formatter('%(asctime)s - %(levelname)s - %(message)s')
+warning_handler = logging.FileHandler(WARN_LOG_FILENAME)
+warning_handler.setLevel(level=logging.WARNING)
+warning_handler.setFormatter(formatter)
+debug_handler = logging.FileHandler(DEBUG_LOG_FILENAME)
+debug_handler.setLevel(level=logging.DEBUG)
+debug_handler.setFormatter(formatter)
+logger.setLevel(logging.DEBUG)
+logger.addHandler(warning_handler)
+logger.addHandler(debug_handler)
+
+
+#HELPER FUNCTIONS
+def split_into_sentences(text):
+    alphabets= "([A-Za-z])"
+    prefixes = "(Mr|St|Mrs|Ms|Dr)[.]"
+    suffixes = "(Inc|Ltd|Jr|Sr|Co)"
+    starters = "(Mr|Mrs|Ms|Dr|He\s|She\s|It\s|They\s|Their\s|Our\s|We\s|But\s|However\s|That\s|This\s|Wherever)"
+    acronyms = "([A-Z][.][A-Z][.](?:[A-Z][.])?)"
+    websites = "[.](com|net|org|io|gov)"
+    text = " " + text + "  "
+    text = text.replace("\n"," ")
+    text = re.sub(prefixes,"\\1<prd>",text)
+    text = re.sub(websites,"<prd>\\1",text)
+    if "Ph.D" in text: text = text.replace("Ph.D.","Ph<prd>D<prd>")
+    text = re.sub("\s" + alphabets + "[.] "," \\1<prd> ",text)
+    text = re.sub(acronyms+" "+starters,"\\1<stop> \\2",text)
+    text = re.sub(alphabets + "[.]" + alphabets + "[.]" + alphabets + "[.]","\\1<prd>\\2<prd>\\3<prd>",text)
+    text = re.sub(alphabets + "[.]" + alphabets + "[.]","\\1<prd>\\2<prd>",text)
+    text = re.sub(" "+suffixes+"[.] "+starters," \\1<stop> \\2",text)
+    text = re.sub(" "+suffixes+"[.]"," \\1<prd>",text)
+    text = re.sub(" " + alphabets + "[.]"," \\1<prd>",text)
+    if "”" in text: text = text.replace(".”","”.")
+    if "\"" in text: text = text.replace(".\"","\".")
+    if "!" in text: text = text.replace("!\"","\"!")
+    if "?" in text: text = text.replace("?\"","\"?")
+    text = text.replace(".",".<stop>")
+    text = text.replace("?","?<stop>")
+    text = text.replace("!","!<stop>")
+    text = text.replace("<prd>",".")
+    sentences = text.split("<stop>")
+    sentences = sentences[:-1]
+    sentences = [s.strip() for s in sentences]
+    return sentences
+
+
+#CLASSES
+class EmbeddingType(Enum):
+    SPECIES = KEY_SPECIES
```

Comment:
As far as I can tell  you set key species as the string species then set species equal to key_species. Whats the point of this and why not just set species = 'species'?